### PR TITLE
Add DECIMAL divide function

### PR DESCRIPTION
**Pull request by karteekmurthys**

### CODE REVIEW

File: velox/functions/prestosql/tests/CMakeLists.txt [hunk 1]
```diff
@@ -19,54 +19,55 @@ target_link_libraries(velox_functions_test_lib velox_exec velox_exec_test_util
...
+  # SliceTest.cpp
+  # SplitTest.cpp
+  # StringFunctionsTest.cpp
+  # TransformTest.cpp
+  # TransformKeysTest.cpp
+  # TransformValuesTest.cpp
+  # URLFunctionsTest.cpp
+  # WidthBucketArrayTest.cpp
+  # GreatestLeastTest.cpp
+  # ZipTest.cpp)
```
Comment (lingbin) [inline velox/functions/prestosql/tests/CMakeLists.txt:69]: It looks like these unit tests have been incorrectly commented out.

**Comment by majetideepak** [comment]
@karteekmurthys can you rebase with main?

File: velox/common/base/CheckedArithmetic.h [hunk 2]
```diff
@@ -96,9 +96,7 @@ inline UnscaledShortDecimal checkedMultiply(
       __builtin_mul_overflow(a.unscaledValue(), b.unscaledValue(), &result);
   if (UNLIKELY(overflow)) {
```
Comment (majetideepak) [inline velox/common/base/CheckedArithmetic.h:97]: Do we not need `valueInRange` check here?

Comment (karteekmurthys) [reply velox/common/base/CheckedArithmetic.h:97]: This PR https://github.com/facebookincubator/velox/pull/2494/files#diff-ba972f0854a87c036d8b3f4e749e3936b0e74fbd0fa4f3c92a647e99d94f2c5bR136 covers it.

**Comment by karteekmurthys** [comment]
@kgpai would you please review these changes?

**Review by majetideepak — APPROVED** [review]

**Review by kgpai — APPROVED** [review]

File: velox/type/UnscaledShortDecimal.h [hunk 3]
```diff
@@ -91,6 +104,19 @@ struct UnscaledShortDecimal {
     return UnscaledShortDecimal(static_cast<int64_t>(value));
   }
 
+  UnscaledShortDecimal operator/(const UnscaledShortDecimal& other) const {
```
[the remark this answers is not in the document]
Comment (karteekmurthys) [reply velox/type/UnscaledShortDecimal.h:107]: Currently I'm not using a /= operator, so didn't implement one. This should suffice right?

File: velox/functions/prestosql/DecimalArithmetic.cpp [hunk 4]
```diff
@@ -186,6 +186,49 @@ class Multiply {
...
+
+  inline static uint8_t
+  computeRescaleFactor(uint8_t fromScale, uint8_t toScale, uint8_t rScale) {
+    return rScale - fromScale + toScale;
+  }
+
+  inline static std::pair<uint8_t, uint8_t> computeResultPrecisionScale(
+      const uint8_t aPrecision,
+      const uint8_t aScale,
+      const uint8_t bPrecision,
```
Comment (kgpai) [inline velox/functions/prestosql/DecimalArithmetic.cpp:224]: bPrecision not used, can you comment it out ?

File: velox/functions/prestosql/DecimalArithmetic.cpp [hunk 5]
```diff
@@ -186,6 +186,49 @@ class Multiply {
   }
 };
 
+class Divide {
+ public:
+  template <typename R, typename A, typename B>
+  inline static void
+  apply(R& r, const A& a, const B& b, uint8_t aRescale, uint8_t bRescale) {
```
Comment (kgpai) [inline velox/functions/prestosql/DecimalArithmetic.cpp:193]: nit: brescale isnt used, can you comment it out ?

**Comment by kgpai** [comment]
@karteekmurthys Some minor nits, hoping to merge this today.

**Pull request closed without merging**